examples/ref-templates/assets/dynamic-fee.py

```diff
@@ -92,4 +87,31 @@ def dynamic_fee(TMPL_TO,
     return And(required_condition, params_condition)
     
 if __name__ == "__main__":
-    print(compileTeal(dynamic_fee(john, amount, master, first_valid, last_valid, lease), Mode.Signature))
+
+#replace these values with your customized values or pass an external parameter
+  scParam = {
+    "TMPL_TO": "2UBZKFR6RCZL7R24ZG327VKPTPJUPFM6WTG7PJG2ZJLU234F5RGXFLTAKA",
+    "TMPL_AMT": 700000,
+    "TMPL_CLS": "WWYNX3TKQYVEREVSW6QQP3SXSFOCE3SKUSEIVJ7YAGUPEACNI5UGI4DZCE",
+    "TMPL_FV": 10,
+    "TMPL_LV": 1000000,
+    "TMPL_LEASE": "023sdDE2"
+  }
+
+  # decode external parameter and update current values.
+  # (if an external paramter is passed)
+  if(sys.argv[1] != "\n"):
+    try:
+      param = yaml.safe_load(sys.argv[1])
+      for key, value in param.items():
+        scParam[key] = value
+    except yaml.YAMLError as exc:
+      print(exc)
```

Comment:
1. Why `sys.argv[1]` will thrown an exception if no parameter is provided. Also, why do we check against `"\n"`?
1. let's create a function for that: `parseArgs`, we can put it in `examples/algobpy` (since this will be a python package, we will need to create an empty `__init__.py` file there to make sure we can import it in other places), and use it in all PyTEAL scripts. The function can return an object or `None`. If object is returned we can overwrite a default object.
1. let's use the function above in all other PyTEAL scripts.

Reply:
1. when using `spawanSync ` without arg 
`spawnSync('python3', [path.join(ASSETS_DIR, filename) ], {..} );`
argv[0] = scripts/name and argv[1] becomes `\n` line
that's why it doesn't throw exception and using "\n". I checked by running both the cases.

2., 3.  Ok

Reply:
For 1. I was doing YAML.stringify() before checking if scInitParam is undefined, that's why it was converted into "\n"

Adding a check.